Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -1034,39 +1044,198 @@ function Set-ClusterDefaultStoragePolicy {
         foreach ($Datastore in $ClusterDatastores) {
             if ($Datastore -in $CompatibleDatastores) {
                 try {
                     Write-Host "Setting Storage Policy on $Datastore to $StoragePolicyName..."
                     Set-SpbmEntityConfiguration -Configuration (Get-SpbmEntityConfiguration $Datastore) -storagePolicy $StoragePolicy -ErrorAction Stop -Confirm:$false
                     Write-Output "Successfully set the storage policy on $Datastore to $StoragePolicyName"
-                } catch {
+                }
+                catch {
                     Write-Error "Was not able to set the storage policy on the Cluster Datastore: $($PSItem.Exception.Message)" -ErrorAction Stop
                 }
-            } else {
+            }
+            else {
                 Write-Error "Modifying the default storage policy on $Datastore is not supported" -ErrorAction Continue
                 continue
             }
         }
     }
 }
+<#
+    .Synopsis
+    Restarts the HCX Manager VM
+    .Parameter Force
+    Flag to force the restart of the hcxmanager without checking for power state, migrations, or replications.
+    For example, A stuck migration could be preventing the restart without this parameter.
+    .Parameter HardReboot
+    Warning: This Parameter should be used as a last ditch effort where a soft-reboot wouldn't work.
+    Hard Reboots the VM instead of restarting the Guest OS.
+    .Parameter Timeout
+    Number of seconds the script is allowed to wait for sucessful connection to the hcx appliance before timing out.
+    .Example
+    # Skips Migrations and replications and hard reboots the system.
+    Restart-HcxManager -Force -HardReboot
+#>
+function Restart-HCXManager {
+    Param(
+        [parameter(
+            Mandatory = $false,
+            HelpMessage = "Force restart without checking for migrations")]
+        [switch]
+        $Force,
+        [Parameter(
+            Mandatory = $false,
+            HelpMessage = 'Hard Reboots the VM')]
+        [ValidateNotNull()]
+        [switch]
+        $HardReboot,
+        [Parameter(
+            Mandatory = $false,
+            HelpMessage = 'Timeout for connection to HCX Server')]
+        [ValidateNotNull()]
+        [int]
+        $Timeout = 600
+    )
+    try {
+        $DefaultViConnection = $DefaultVIServers
+        $UserName = 'tempHcxAdmin'
+        $UserRole = 'tempHcxAdminRole'
+        $Group = 'CloudAdmins'
+        $Port = 443
+
+        Write-Host "Creating new temp scripting user"
+        $privileges = @("VirtualMachine.Interact.PowerOff",
+            "VirtualMachine.Interact.PowerOn")
+        $HcxAdminCredential = New-TempUser -privileges $privileges -userName $UserName -userRole $UserRole
+
+        Write-Host "INPUTS: HardReboot=$HardReboot, Force=$Force, Port=$Port, Timeout=$Timeout"
+
+        $HcxServer = 'hcx'
+        $hcxVm = Get-HcxManagerVM
+
+        Add-UserToGroup -userName $UserName -group $Group
```

Comment:
I don't believe belonging to CloudAdmins will let this user access our management resources.